lib/file-finder.js

```diff
@@ -114,27 +116,29 @@ class FileFinder {
                         for (let j = 0; j < searched; j++) {
                             cache[dirs[j]].push(filePath);
                         }
-
+                        yield filePath;
                         break;
                     }
                 }
             }
+
             const child = directory;
 
             // Assign parent directory to directory.
             directory = path.dirname(directory);
 
             if (directory === child) {
-                return cache[dirs[0]];
+                yield* cache[dirs[0]];
```

Comment:
I think this should just be `return;`, because by this point all of the file paths in the cache will have already been `yield`-ed, so it's not necessary to `yield` them again. (This will also avoid the need to use `.slice` to make the tests pass.)